src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
+
+@ApiTags('Auth')
+@Controller()
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get('oauth/callback')
+  async githubAuthCallback(@Query('code') code: string) {
+    try {
+      const jwtToken = await this.authService.handleGitHubCallback(code);
+
+      return jwtToken;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
```

Comment:
PullProofAI


The error handling in the `githubAuthCallback` method can be improved. Instead of checking `error.status`, which might not always be present, it's better to handle errors based on the type of error received. Consider revising the error handling logic to provide more specific error messages and HTTP statuses.

```diff
-       if (error.status) {
-         throw new HttpException(error.message, error.status);
+       if (error instanceof HttpException) {
+         throw error;
+       } else {
+         throw new HttpException('Failed to handle GitHub callback', HttpStatus.INTERNAL_SERVER_ERROR);
```



<!-- This is an auto-generated comment by PullProofAI -->